scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+
+        In terms of probability density function :math:`f(x)` and its support
+        :math:`\chi`, the differential entropy of a random variable :math:`X` is:
+
+        .. math::
+
+            h(X) = - \int_{\chi} f(x) \log f(x) dx
+
+        Parameters
+        ----------
+        method : {None, 'formula', 'logexp', 'quadrature'}
+            The strategy used to evaluate the differential entropy. By default
+            (``None``), the infrastructure chooses between the following options.
+
+            - ``'formula'``: use a formula for the differential entropy itself
+            - ``'logexp'``: evaluate the logarithm of the differential entropy
+                            directly and exponentiate
+            - ``'quadrature'``: numerically integrate the PDF
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The differential entropy of the random variable.
+
+        See Also
+        --------
+        ContinuousDistribution.logentropy
+        ContinuousDistribution.pdf
+
+        Notes
+        -----
+        This function calculates the differential entropy using the natural
+        logarithm; i.e. the logarithm with base :math:`e`. Consequently, the
+        value is expressed in (dimensionless) "units" of nats. To convert the
+        entropy to different units (i.e. corresponding with a different base),
+        divide the result by the natural logarithm of the desired base.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-1., b=1.)
+
+        Evaluate the differential entropy:
+
+        >>> X.entropy()
+        0.6931471805599454
+
+        """
+        return self._entropy_dispatch(method=method, **self._parameters)
+
+    @_dispatch
+    def _entropy_dispatch(self, method=None, **kwargs):
+        if self._overrides('_entropy_formula'):
+            method = self._entropy_formula
+        elif self._overrides('_logentropy_formula'):
+            method = self._entropy_logexp
+        else:
+            method = self._entropy_quadrature
+        return method
+
+    def _entropy_formula(self, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _entropy_logexp(self, **kwargs):
+        return np.real(np.exp(self._logentropy_dispatch(**kwargs)))
+
+    def _entropy_quadrature(self, **kwargs):
+        def integrand(x, **kwargs):
+            pdf = self._pdf_dispatch(x, **kwargs)
+            return np.log(pdf)*pdf
+        return -self._quadrature(integrand, kwargs=kwargs)
+
+    @_set_invalid_nan_property
+    def median(self, *, method=None):
+        r"""Median
+
+        If a continuous random variable :math:`X` has probability :math:`0.5` of
```

Comment:
this is not correct (you can say that m is the largest such value).

Reply:
Do we have any _continuous_ distributions in SciPy for which this is incorrect as written?

I can see the problem if we were talking about a discrete distribution or if we had distributions for which the PDF drops to zero near the median. Is there another case I'm not considering?

I don't want to give a mathematics textbook definition of a median. It should be readable by the majority of readers (who don't care about the special cases) and strategically vague to avoid being deemed *incorrect* by statistics PhDs.

Can you think of a way to walk the line here _without adding complexity_?

Reply:
you can resolve this one, I misread the statement